### ISSUE TITLE
Make packages tree read-only

### ISSUE THREAD

**Issue by DartBot** [dart-lang/pub#419]
<a href="https://github.com/stevemessick"><img src="https://avatars.githubusercontent.com/u/8518285?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [stevemessick](https://github.com/stevemessick)**
_Originally opened as dart-lang/sdk#8101_

----

Enhancement suggestion: Make editor prevent you from changing anything in the packages tree.

I had incorrectly set up a package I was trying to create, by storing my source code under a subfolder of the packages tree.  This &quot;got it to work&quot;, but was not the right way of doing it.

When I upgraded to the latest editor and loaded up my project, it (pub package-manager?) assassinated the source files I had created.

Luckily I had backups, so no damage done.  The folks in the IRC channel suggested I report this feedback to help keep others from similarly &quot;shooting themselves in the foot&quot;

From: rk22a in IRC channel / richardk@leapbeyond.com
////////////////////////////////////////////////////////////////////////////////////
Editor: 0.3.1_r17463 (2013-01-23)
OS: Windows 7 - amd64 (6.1)
JVM: 1.6.0_21
# projects: 1
# open dart files: 2
auto-run pub: true
mem max/total/free: 889 / 580 / 459 MB
thread count: 24
analysis: 0 tasks, 35 libraries, 2 contexts
index: 80337 relationships and 0 attributes in 15398 elements in 148 resources stored in 2 MB on disk

SDK installed: true
Dartium installed: true

**Comment by DartBot** [comment]
<a href="https://github.com/devoncarew"><img src="https://avatars.githubusercontent.com/u/1269969?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [devoncarew](https://github.com/devoncarew)**

----

I think a good way to implement this would be for the files to be read-only on disk. i.e. for pub to make them read-only during provisioning. That way there's no special casing needed by the editor, and other development tools (IntelliJ, Sublime, ...) will see the same behavior (and avoid the same issues).

**Comment by DartBot** [comment]
<a href="https://github.com/keertip"><img src="https://avatars.githubusercontent.com/u/2192312?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [keertip](https://github.com/keertip)**

----

Good point! For now, we have a fix in the editor to open as read only

https://codereview.chromium.org/12194031/

r=18100

It would be great if pub marked the files as readonly when installing them in pub cache
______
*Set owner to @munificent.*
*Removed this from the [M4](../milestones/M4) milestone.*
*Removed [Area-Editor](../labels/Area-Editor) label.*
*Added [Area-Pub](../labels/Area-Pub) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

*Added this to the [Later](../milestones/Later) milestone.*

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

*Marked this as being blocked by dart-lang/sdk#15078.*

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

I don't think we intend to handle this in file system. Hopefully we can get rid of symlinks entirely and sidestep this issue.
______
*Added [NotPlanned](../labels/NotPlanned) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

Issue dart-lang/sdk#16263 has been merged into this issue.
